Python/ibmcloudsql/catalog_table.py

```diff
@@ -0,0 +1,311 @@
+                                              cos_url,
+                                              force_recreate=force_recreate)
+        if job_id is not None and blocking is True:
+            job_status = self.wait_for_job(job_id)
+            if job_status == "completed":
+                return self.get_result(job_id)
+            else:
+                return None
+        return None
+
+    def get_catalog_table_partitioned(self, table_name, cos_url=None, force_recreate=False):
+        """
+        Create a partitioned catalog table. We need to call `refresh_cache_table_partitioned`
+
+        Parameters
+        --------------
+        table_name: str
+            the name of the catalog table to be created
+
+        cos_url : str, optional
+            The COS URL from which the catalog table should reference to
+            If not provided, it uses the internal `self.cos_in_url`
+
+        force_recreate: bool
+            (True) force to recreate an existing catalog table
+
+        Returns
+        ----------
+
+        """
+        self.current_catalog_table_name = table_name
+
+        if cos_url is None:
+            cos_url = self.cos_in_url_partitioned
+
+        df = self.show_catalog_tables()
+        try:
+            found = df[df['tableName'].str.contains(table_name)]
+        except Exception:
+            # not found
+            found = []
+        if len(found) > 0 and force_recreate:
+            self.drop_catalog_table(table_name)
+        self.partitioned_tables.add(table_name)
+        if len(found) == 0 or force_recreate:
+            if True:
+                # auto-detection of scheme
+                self.sql_stmt_create_partitioned_template = """
+                CREATE TABLE {table_name}
+                USING CSV
+                LOCATION {cos_in}
+                """
+                sql_stmt_create_partitioned = self.sql_stmt_create_partitioned_template.format(
+                    table_name=table_name, cos_in=cos_url)
+            else:
+                # explit selection of scheme -> need to tell "PARTITIONED BY"
+                sql_stmt_create_partitioned = """
+                CREATE TABLE {table_name} (
+                customerID string,
+                companyName string,
+                contactName string,
+                contactTitle string,
+                address string,
+                region string,
+                postalCode string,
+                country string,
+                phone string,
+                fax string
+                )
+                USING CSV
+                PARTITIONED BY (country)
+                LOCATION {cos_in}
+                """.format(table_name=table_name, cos_in=cos_url)
+            logger.debug(sql_stmt_create_partitioned)
+            self.run_sql(sql_stmt_create_partitioned)
+            time.sleep(2)
+            self.refresh_cache_table_partitioned(table_name)
+
+    def refresh_cache_table_partitioned(self, table_name):
```

Comment:
It is not a cache being affected here. Simply name the method after the Hive operation: recover_table_partitions

Reply:
Updated in commit `00fc7b5bd48668386d01dad60f89b1069c8f8f96`